packages/ui5-middleware-onelogin/src/cookieGetter.ts

```diff
@@ -135,10 +168,38 @@ export default class CookieGetter {
 
 		try {
 			const browser = await chromium.launch(playwrightOpt);
-			const context = await browser.newContext({ ignoreHTTPSErrors: true });
+			const contextOptions: any = { ignoreHTTPSErrors: true };
+
+			if (useClientCertificates) {
+				contextOptions.clientCertificates = [
+					{
+						origin: effectiveOptions.configuration.certificateOrigin,
+						certPath: effectiveOptions.configuration.certificateCertPath,
+						cert: effectiveOptions.configuration.certificateCert,
+						keyPath: effectiveOptions.configuration.certificateKeyPath,
+						key: effectiveOptions.configuration.certificateKey,
+						pfxPath: effectiveOptions.configuration.certificatePfxPath,
+						pfx: effectiveOptions.configuration.certificatePfx,
+						passphrase: effectiveOptions.configuration.certificatePassphrase,
+					},
+				];
+			}
+
+			if (effectiveOptions.configuration.debug) {
+				log.info("Client certificates configuration:");
+				// Create a copy of certificates config without the passphrase for logging
+				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-return
+				const certificatesForLogging = contextOptions.clientCertificates.map((cert: any) => ({
```

Comment:
My first thought was that you could maybe reuse the `sanitizePassphrase` function here but I guess since you don't know the object type and thus the path of where you'd expect the passphrase to be, at least in its current implementation, you cannot reuse this. Maybe it can be an idea later on to change this.

Reply:
yeah, different objects, bit more difficult